src/pages/Option/Index.js

```diff
@@ -37,9 +43,73 @@ const StyledDrawer = styled(Drawer)`
     color: #b5b5b6;
   }
 `
+   
+function safeDiv(a, b) {
+    return b==0 ? 0 : a.div(b).toNumber();
+}
+
+export function getPosition(rawPositionData,symbolData) {
+  if (!rawPositionData || !symbolData) {
+    return
+  }
+  let positionQuery = []
+  for(let i = 0; i < rawPositionData.positions.length; i++){
+    const temp = rawPositionData.positions[i]
+    const markPrice = symbolData.find(obj => {
+      return obj.address === temp[0]
+    }).markPrice
+    const initialMarginRatio = symbolData.find(obj => {
+      return obj.address === temp[0]
+    }).initialMarginRatio
+    const indexPrice = symbolData.find(obj => {
+      return obj.address === temp[0]
+    }).indexPrice
+    const marginUsage = Math.abs(temp[2].toNumber() * indexPrice) * initialMarginRatio
+    const unrealizedPnl = temp[2].toNumber() * indexPrice - temp.cost
+    const position = {
+      symbol: temp[1],
+      address: temp[0],
+      position: temp[2].toNumber(),
+      entryPrice: safeDiv(temp.cost,temp[2]),
+      markPrice: symbolData.find(obj => {
+        return obj.address === temp[0]
+      }).markPrice,
+      marginUsage: symbolData.find(obj => {
+        return obj.address === temp[0]
+      }).initialMarginRatio,
+      unrealizedPnl: symbolData.find(obj => {
+        return obj.address === temp[0]
+      }).indexPrice,
+      accountFunding: temp.cumulativeFundingPerVolume.toNumber() * temp[2].toNumber(),
+      type:temp[2].toNumber()>=0?"Long":"Short",
+    };
+    positionQuery.push(position)
+  }
+  return positionQuery;
+}
+
+export function getSymbol(rawSymbolData){
+  if (!rawSymbolData) {
+    return
+  }
+  let symbolQuery = []
+  for(let i = 0; i < rawSymbolData.length; i++){
+    const temp = rawSymbolData[i]
+    const symbol = {
+      symbol: temp[1],
+      address: temp[2],
+      markPrice: temp.markPrice.div(bigNumberify('1000000000000000000')).toNumber(),
+      indexPrice: temp.indexPrice.div(bigNumberify('1000000000000000000')).toNumber(),
+      initialMarginRatio: temp.initialMarginRatio.div(bigNumberify('1000000000000000000')).toNumber(),
+    };
+    symbolQuery.push(symbol)
+  }
+  return symbolQuery;
+}
 
 const Option = props => {
-  const { account, library } = useConstantLoader();
+  const { account, library } = useConstantLoader(props);
```

Comment:
We are deprecating the use of `useConstantLoader` now, only for `chainId` we get it from `useChainId()`, all other ethers objects we get them via `{ account, signer, library, ... } = useWeb3React()`